microsoft-edge/webview2/samples/webview2-winui3-sample.md

```diff
@@ -0,0 +1,603 @@
+        public App()
+        {
+            this.InitializeComponent();
+            // If you're shipping a fixed-version WebView2 Runtime with your app,
+            // un-comment the following code and change the version number to the
+            // version number of the WebView2 Runtime that you're packaging and
+            // shipping to users:
+            // StorageFolder localFolder = Windows.ApplicationModel.Package.Current.InstalledLocation;
+            // String fixedPath = Path.Combine(localFolder.Path, "FixedRuntime\\95.0.1020.53");
+            // Debug.WriteLine($"Launch path [{localFolder.Path}]");
+            // Debug.WriteLine($"FixedRuntime path [{fixedPath}]");
+            // Environment.SetEnvironmentVariable("WEBVIEW2_BROWSER_EXECUTABLE_FOLDER", fixedPath);
+        }
+
+        /// <summary>
+        /// Invoked when the application is launched normally by the end user.
+        /// </summary>
+        /// <param name="args">Details about the launch request and process.</param>
+        protected override void OnLaunched(Microsoft.UI.Xaml.LaunchActivatedEventArgs args)
+        {
+            m_window = new MainWindow();
+            m_window.Activate();
+        }
+
+        private Window m_window;
+    }
+}
+```
+
+
+<!-- ------------------------------ -->
+#### MainWindow.xaml
+
+
+<!-- ---------- -->
+###### XML elements
+
+| Element | Control |
+|---|---|
+| `<TextBox>` | Address bar |
+| `<Button>` | **Go** button |
+| `<WebView2>` | WebView2 control |
+| `<TextBlock>` | Status bar |
+
+
+<!-- ---------- -->
+###### Condensed listing
+
+```xaml
+<Window
+    xmlns:local="using:WebView2_WinUI3_Sample">
+    <Grid>
+        <Grid.RowDefinitions>
+        </Grid.RowDefinitions>
+        <Grid Grid.Row="0" Background="LightGray">
+            <TextBox Grid.Column="0"  x:Name="AddressBar" KeyDown="AddressBar_KeyDown" VerticalAlignment="Center" Margin="10,0,0,0"/>
+            <Button Grid.Column="1" x:Name="Go" Content="Go" Click="Go_OnClick" VerticalAlignment="Center" HorizontalAlignment="Center"/>
+        </Grid>
+
+        <WebView2 x:Name="WebView2" Grid.Row="1"/>
+
+        <Rectangle Grid.Row="2" Fill="LightGray"/>
+        <TextBlock x:Name="StatusBar" Text="WebView2" VerticalAlignment="Center" Grid.Row="2" Margin="10,0,10,0"/>
+    </Grid>
+</Window>
+```
+
+<!-- ------------------------------ -->
+#### MainWindow.cs
+
+
+<!-- ---------- -->
+###### Methods
+
+* `WebView2_WinUI3_Sample` namespace
+   * `MainWindow` class
+      * `StatusUpdate(message)`
+      * `WebView2_CoreWebView2Initialized(InitializedEventArgs)`
+      * `WebView2_NavigationCompleted(NavigationCompletedEventArgs)`
+      * `TryCreateUri(potentialUri, out uri)`
+      * `TryNavigate()`
+      * `Go_OnClick(sender, RoutedEventArgs)`
+      * `AddressBar_KeyDown(sender, KeyRoutedEventArgs)`
+      * `SetTitle()`
+      * `GetWebView2Version()`
+
+
+<!-- ---------- -->
+###### Condensed listing
+<!-- todo: trim listing -->
```

Comment:
i wish to condense this code @captainbrosset

Reply:
doc-design idea:
* insert h6 heading between each method, creating a spot to add commentary

Reply:
closing.  i put a subhead between each method